packages/nexus-sdk/src/File/index.ts

```diff
@@ -1,22 +1,192 @@
-import { toPromise } from '@bbp/nexus-link';
+import FormData from 'isomorphic-form-data';
+import { toPromise, FetchAs } from '@bbp/nexus-link';
+import {
+  GetFileOptions,
+  NexusFile,
+  FilePayload,
+  LinkFilePayload,
+  UpdateFilePayload,
+} from './types';
+import { buildQueryParams, isBrowser } from '../utils';
+import {
+  PaginatedResource,
+  ResourceListOptions,
+  TagResourcePayload,
+  GetResourceOptions,
+} from '../Resource/types';
+import { Observable } from '../../../nexus-link/lib';
 
-const NexusFile = ({ httpGet }, context: any) => {
+const NexusFile = (
+  { httpGet, httpPost, httpPut, httpDelete, poll },
+  context: any,
+) => {
   return {
     get: (
       orgLabel: string,
       projectLabel: string,
       fileId: string,
-    ): Promise<Blob> =>
-      toPromise(
+      options?: GetFileOptions,
+    ): Promise<NexusFile | Blob | string | FormData> => {
+      const { as = FetchAs.BLOB, ...opts } = options;
+      return toPromise(
         httpGet({
           path: `${context.uri}/${
             context.version
-          }/files/${orgLabel}/${projectLabel}/${fileId}`,
+          }/files/${orgLabel}/${projectLabel}/${fileId}${buildQueryParams(
+            opts,
+          )}`,
           context: {
-            as: 'blob',
+            as,
           },
         }),
+      );
+    },
+
+    list: (
+      orgLabel: string,
+      projectLabel: string,
+      options?: ResourceListOptions,
+    ): Promise<PaginatedResource<NexusFile>> => {
+      const opts = buildQueryParams(options);
+      return toPromise(
+        httpGet({
+          path: `${context.uri}/${
+            context.version
+          }/files/${orgLabel}/${projectLabel}${opts}`,
+        }),
+      );
+    },
+
+    create: (
+      orgLabel: string,
+      projectLabel: string,
+      payload: FilePayload,
+    ): Promise<NexusFile> => {
+      const { '@id': fileId, file, storage } = payload;
+      const body = new FormData();
```

Comment:
I would honestly let the client deal with the data they want to upload. The way I see it, it should be like:

```javascript
// some logic in the browser
const data = new FormData();
data.append('foo', 'bar');
data.append('file', document.getElementById('file').files[0]);
// now use sdk
nexus.File.create('myorg', 'myproject', { file: data });
```

```javascript
// some logic in node
const FormData = require('form-data');

const data = new FormData()
data.append('foo', 'bar');
data.append('file', fs.createReadStream(__dirname + '/file.txt'));
// now use sdk
nexus.File.create('myorg', 'myproject', { file: data });
```